tests/drivers/can/api/src/classic.c

```diff
@@ -516,6 +534,47 @@ ZTEST_USER(can_classic, test_invalid_sample_point)
 	zassert_equal(err, -EINVAL, "invalid sample point of 100.0% accepted (err %d)", err);
 }
 
+/**
+ * @brief Test using an invalid bitrate.
+ */
+ZTEST_USER(can_classic, test_invalid_bitrate)
+{
+	struct can_timing timing;
+	int err;
+
+	err = can_calc_timing(can_dev, &timing, 0, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 0 bit/s accepted (err %d)", err);
+
+	err = can_calc_timing(can_dev, &timing, 2000000, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 2Mbit/s accepted (err %d)", err);
```

Comment:
I am not sure this warrants an API test.

Reply:
Value of 1Mbit/s is taken from CAN specification.
I will change from 2000000 to 1000001 to test edge condition..